src/pipdeptree/_render/text.py

```diff
@@ -59,7 +59,16 @@ def aux(  # noqa: PLR0913, PLR0917
         has_grand_parent: bool = False,  # noqa: FBT001, FBT002
         is_last_child: bool = False,  # noqa: FBT001, FBT002
         parent_is_last_child: bool = False,  # noqa: FBT001, FBT002
+        visited_nodes=None,
     ) -> list[Any]:
+        if visited_nodes is None:
+            visited_nodes = set()
+
+        if node in visited_nodes:
+            return []
+
+        visited_nodes.add(node)
+
```

Comment:
No need to add a visited set as `cur_chain` makes sure that we don't end up revisiting the same packages when transversing a top-level package's dependencies.

Reply:
Oh, sorry. I didn't noticed `cur_chain`. You are right, `cur_chain` is all we need. I will remove this.